### ISSUE TITLE
Guidance around persisting Data Explorer metadata

### ISSUE THREAD

**Issue by dylanross3** [nteract/nteract#4304]
Using the Data Explorer component, the `onMetadataChange` prop is called when the user changes the selected UI configuration of the component (such as by switching the chart type). It appears that the nteract notebook UI persists this metadata in a `dx` key in root of the output's metadata within the notebook file.

```
notebook.cells[0].outputs[0].metadata.dx
```

**My question is:** Is this the recommended place to store the Data Explorer metadata, or would it be better to store the metadata under the MIME type considering it applies to only that part of the output?

```
notebook.cells[0].outputs[0].metadata["application/vnd.dataresource+json"].dx
```

I don't know if there is any practical need for this, but it seemed like something to consider after seeing the following example.

https://nbformat.readthedocs.io/en/latest/format_description.html#display-data
```
"metadata" : {
  "image/png": {
    "width": 640,
    "height": 480,
  },
},
```

**Comment by emeeks** [comment]
This is a better question for @rgbkrk than me but I thought it was working this way (that the metadata was scoped to the mediaType).

**Comment by rgbkrk** [comment]
You're correct @dylanross3, the metadata needs to be stored under the `application/vnd.dataresource+json` key. The current implementation is incorrect, as it needs to be scoped to the right media type.

**Comment by dylanross3** [comment]
Thanks for the response.

1. Do we still want the `dx` key in case there is other metadata relevant to the `"application/vnd.dataresource+json"` MIME type?

```
"metadata": {
  "application/vnd.dataresource+json": {
    "dx": {
      view: "line"
    }
  }
}
```

2. Should we make this change in a backwards compatible way, so the Data Explorer will initialize with metadata in the old location if not present in the new location? It could also remove the old metadata when calling `onMetadataChange`. I need the backwards compatibility for old notebooks, but I could implement this on my end if we don't want it in the Data Explorer package.

3. While we are changing metadata, do we want to make other changes such as to allow multiple configurations for #4385? Not sure if there is interest in that issue. If not, I can easily implement it in user land and add an additional metadata field to keep track of the extra configurations. If possible, I'd like to keep maximum portability of a notebook between nteract and my company's custom notebook UI, so I don't want to stray too much from what you guys are doing.

**Comment by rgbkrk** [comment]
> Do we still want the dx key in case there is other metadata relevant to the `application/vnd.dataresource+json` MIME type?

I think scoping it to a `dx` key is smart so that we don't have to worry about stepping on the toes of anyone else using this media type.

> Should we make this change in a backwards compatible way, so the Data Explorer will initialize with metadata in the old location if not present in the new location? It could also remove the old metadata when calling onMetadataChange. I need the backwards compatibility for old notebooks, but I could implement this on my end if we don't want it in the Data Explorer package.

I feel like adoption is likely nascent enough with changes occurring frequently that we should just switch to the new one (and possibly convert from the old on notebook load). Within the `dx` python package I was scoping the metadata originally: https://github.com/nteract/dx/blob/master/dx/__init__.py

For number 3, I'll have to come back to thinking about that one.

**Comment by captainsafia** [comment]
Transferring to data explorer repo.
